packages/dds/migration-shim/src/types.ts

```diff
@@ -21,3 +27,14 @@ export interface IShimDeltaHandler extends IDeltaHandler {
 	 */
 	hasTreeDeltaHandler(): boolean;
 }
+
+/**
+ * An interface for a shim channel that intercepts a LegacySharedTree or new SharedTree DDS.
+ *
+ * @internal
+ */
+export interface IShim extends IChannel {
```

Comment:
Probably don't need this interface since we don't intend to generalize further at this point, would be more obvious what's going on to just use typing of `MigrationShim | SharedTreeShim`

Reply:
I did this specifically for typing, since I wanted `ShimHandle<MigrationShim>` in `MigrationShim` and `ShimHandle<SharedTreeShim>` in `SharedTreeShim`